src/common/HTTPRequest.ts

```
@@ -36,6 +36,14 @@ export interface ContinueRequestOverrides {
   headers?: Record<string, string>;
 }
 
+/**
+ * @public
+ */
+export interface InterceptResolutionState {
+  strategy: InterceptResolutionStrategy;
```

Since you are tweaking this, may I suggest to name this an interception `action` instead of a `strategy`? 🤔 

When I was implementing Cooperative Mode in `puppeteer-extra`, I found this concept a bit odd when referring to taking an action in the context of intercepting the current request.

I would imagine a strategy to reflect the overall effect of the interceptor function, like "take action `abort` for all the requests with `resourceType = script`", rather than `abort/continue/respond`. What do you think?

---

🤯  THANK YOU! I've been trying to think of the right word for a long time. Changed.

---

Hahaha nice! Easiest sell ever 😄